packages/contracts/contracts/L1/L2OutputOracle.sol

```diff
@@ -0,0 +1,120 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.10;
+
+import { Ownable } from "../../lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+
+/**
+ * @title L2OutputOracle
+ */
+// The payable keyword is used on appendL2Output to save gas on the msg.value check.
+// slither-disable-next-line locked-ether
+contract L2OutputOracle is Ownable {
+    event l2OutputAppended(bytes32 indexed _l2Output, uint256 indexed _l2timestamp);
+
+    // The interval in seconds at which checkpoints must be submitted.
+    uint256 public immutable submissionInterval;
+
+    // The time between blocks on L2.
+    uint256 public immutable l2BlockTime;
+
+    // The number of blocks in the chain before the first block in this contract.
+    uint256 public immutable historicalTotalBlocks;
+
+    // The timestamp of the first L2 block recorded in this contract.
+    uint256 public immutable startingBlockTimestamp;
+
+    // The timestamp of the most recent L2 block recorded in this contract.
+    uint256 public latestBlockTimestamp;
+
+    // A mapping from L2 timestamps to the output root for the block with that timestamp
+    mapping(uint256 => bytes32) public l2Outputs;
+
+    /**
+     * Initialize the L2OutputOracle contract.
+     * @param _submissionInterval The desired interval in seconds at which
+     *        checkpoints must be submitted.
+     * @param _l2BlockTime The desired L2 inter-block time in seconds.
+     * @param _genesisL2Output The initial L2 output of the L2 chain.
+     * @param _historicalTotalBlocks The number of blocks that preceding the
+     *        initialization of the L2 chain.
+     */
+    constructor(
+        uint256 _submissionInterval,
+        uint256 _l2BlockTime,
+        bytes32 _genesisL2Output,
+        uint256 _historicalTotalBlocks,
+        address sequencer
+    ) {
+        submissionInterval = _submissionInterval;
+        l2BlockTime = _l2BlockTime;
+        l2Outputs[block.timestamp] = _genesisL2Output; // solhint-disable not-rely-on-time
+        historicalTotalBlocks = _historicalTotalBlocks;
+        latestBlockTimestamp = block.timestamp; // solhint-disable not-rely-on-time
+        startingBlockTimestamp = block.timestamp; // solhint-disable not-rely-on-time
+
+        _transferOwnership(sequencer);
+    }
+
+    /**
+     * Accepts an L2 output checkpoint and the timestamp of the corresponding L2
+     * block. The timestamp must be equal to the current value returned by
+     * `nextTimestamp()` in order to be accepted.
+     * This function may only be called by the Sequencer.
+     * @param _l2Output The L2 output of the checkpoint block.
+     * @param _l2timestamp The L2 block timestamp that resulted in _l2Output.
+     * @param _blockhash A block hash which must be included in the current chain.
+     * @param _blocknumber The block number with the specified block hash.
+     */
+    function appendL2Output(
```

Comment:
This change in interface is going to break the go code in this codebase, but I suppose that is ok as long as we note that someplace to fix in the future

Reply:
Good callout. 

I took a stab at it, managed update the makefile generate new bindings, but wasn't able to easily figure out how to provide the sequencer address to the constructor. 

[Here's where I got stuck](https://github.com/ethereum-optimism/optimistic-specs/commit/bdecf17fda04f92818aa48f43f20cfe716a9a76a#diff-310ca044c8a6e5b9faaab0c60de3c0b5f9d6952fbf11fa073a4c9f7f6c3759b8L237-L238).